tests/dispatch/test_http2.py

```diff
@@ -212,3 +213,18 @@ def test_http2_reconnect():
 
         assert resp.status_code == 200
         assert resp.http_version == "HTTP/2"
+
+
+async def test_http2_connection_renewed(backend):
+    """Test new connection is formed when stream ids expire"""
+    server = MockHTTP2Server(app=app, backend=backend)
+    server.conn.HIGHEST_ALLOWED_STREAM_ID = 1
+
+    async with httpx.ConnectionPool(backend=backend) as http:
+        response = await http.request("GET", server.url)
```

Comment:
The mock server has no `.url` attribute, which makes the build fail. I think you can use anything here, since we don't make real requests, so e.g. `"/"`.